client/src/assets/player/peertube-player-manager.ts

```diff
@@ -144,7 +145,7 @@ export class PeertubePlayerManager {
       ])
     }
 
-    const videojsOptions = this.getVideojsOptions(mode, options, p2pMediaLoader)
+    const videojsOptions = await this.getVideojsOptions(mode, options, p2pMediaLoader)
```

Comment:
Instead of changing `client/src/root-helpers/plugins.ts` file, just add a `videojsOptionsFilter` parameter in `PeertubePlayerManagerOptions`.

embed and watch page will create a filter function that send the object to plugins

Reply:
Isn't it more simple to have the filter function on one place instead of duplicating the code?

Reply:
Yes you're right but I'd prefer to not couple the peertube plugin system to the player manager (if we can avoid it)

Reply:
I see. But maybe it opens up for more possibilities to make the plugin system agnostic, I'm thinking about running hooks/filter in ServiceWorker (caching, push notifications), etc,

Reply:
This should open up for #3830 by allowing a hook to be called in the ServiceWorker.

Reply:
What do you think?

Reply:
this is a good idea, I'll update and merge the PR